src/utils/windowUtils.ts

```diff
@@ -41,6 +41,30 @@ export module Utils {
       }
     }
 
+    /**
+     * To be used in the first line of a deprecated method
+     *
+     * @param {string} version The version when the tagged method became obsolete
+     */
+    export function deprecated(version: string) {
+
+      var callingMethod = "";
+
+      try {
+        callingMethod = (<any> new Error).stack
+          .split("\n")
+          .filter((step: string) => step.match(/http/))[1]
+          .trim()
+          .split(/\s|@/)
+          .filter((keyword: string) => !keyword.match(/at/))[0];
```

Comment:
Errr... I have no idea what is happening here.  Please explain?  Probably through comments or some better organization